feat: add radio button native app criteria

marshcooke opened the pull request:
Radio Button native app criteria review checklist:

- [ ] Run the project
- [ ]  Cross compare the old & new side by side, note anything missing
- [ ]  Note if anything is styled oddly or missing or broken
- [ ]  Read the new criteria file, note anywhere in the content where edits or changes are needed

danbivins commented on public/content/documentation/native/controls/radio-button.md:
```diff
@@ -0,0 +1,303 @@
+      - **Note:** Setting a programmatic name while a visible text label exists may cause VoiceOver to duplicate the announcement of the name. If this happens, hide the visible text label from VoiceOver recognization.
+   - Placeholder text is NOT the programmatic name.
+
+   - **UIKit**
```
it looks like you're formatting on the h4s for UIKit, SwiftUI, etc. (and same for Android section) didn't occur like the Button documentation.

marshcooke commented:
Updated h4s that didn't port over in raw HTML; also updated the gherkin section to the updated format/structure, discussed and reviewed with Lindsay.
